quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/Configuration.java

```diff
@@ -188,4 +194,23 @@ private static String getValue(ConfigSource configSource, String name) {
 
         return value;
     }
+
+    public static boolean isOptimized() {
+        return Configuration.getRawPersistedProperty(KC_OPTIMIZED).isPresent();
+    }
+
+    public static void markAsOptimized(Properties properties) {
+        properties.put(Configuration.KC_OPTIMIZED, Boolean.TRUE.toString());
+    }
+
+    public static ConfigValue getCurrentBuiltTimeProperty(String name) {
+        PersistedConfigSource persistedConfigSource = PersistedConfigSource.getInstance();
+
+        try {
+            persistedConfigSource.enable(false);
```

Comment:
Without it, we are not able to resolve default values for build options. For instance, the `db=postgres` option is persisted but not set when issuing a `start`. In this case, we need to resolve the default value (e.g.: db=dev-file) in order to show the differences in configuration.

Iterate over the registered config sources, expect the `PersistedConfigSource`, won't work either because in order to resolve the default values we need to go through MicroProfile configuration in order to resolve default values from our mappers.